flycheck.el

```diff
@@ -4609,6 +4609,28 @@ symbols in the command."
          (seq-map (lambda (arg) (flycheck-substitute-argument arg checker))
                   (flycheck-checker-arguments checker))))
 
+(defun flycheck--process-send-buffer (process)
+  "Send contents of current buffer to PROCESS.
+This is a compatibility wrapper around `process-send-region`: on
+all systems but Windows it simply calls that function.  On
+Windows, it attempts to work around an Emacs bug by sending the
+contents of the buffer in small batches (4096 bytes seems to be
+the threshold above which Emacs just hangs).
+
+See https://debbugs.gnu.org/cgi/bugreport.cgi?bug=22344 for more
+information about the Emacs issue, and see the original issue at
+https://github.com/flycheck/flycheck/issues/794 for information
+about the impact on Flycheck."
+  (save-restriction
+    (widen)
+    (if (eq system-type 'windows-nt)
+        (let ((from (point-min)))
```

Comment:
PS: I don't think that we should have a proper `defcustom` for this, for most users should never ever need to change it, or would want to.  A simple `defvar` should be sufficient, in my opinion.